README.md

```diff
@@ -124,6 +125,47 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
+
+If you are using [Glide](https://glide.sh) or [godep](https://github.com/tools/godep) 
+to manage your Go dependencies, this will manage the non-optional libraries automatically.
+
+There are two ways of lining up dependencies - via the repository version *tags* 
+or via the repository *branches*.
+
+Do not mix dependencies with tags and branches between repositories, it is not supported.
+
+##### Tag Version Dependencies
+
+Using the git tag versions is the preferred way of managing dependencies for newer 
+releases of client-go
+
+| client-go    | [apimachinery](https://github.com/kubernetes/apimachinery) | [apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver)<sup>*</sup> | [code-generator](https://github.com/kubernetes/code-generator)<sup>+</sup> |
```

Comment:
Why do people need both client-go and apiextensions-apiserver at the same time?

Reply:
If you wanted to programmatically access CRDs and regular Kubernetes resources in the same binary.

Reply:
https://github.com/kubernetes/client-go/issues/247

Reply:
`apiextensions-apiserver` is doing it wrong, then, its api needs to be available separately. It is madness to require clients to vendor server code just to get the client.